sovryn-token-bridge/federator/src/lib/GasServices.js

```diff
@@ -0,0 +1,77 @@
+const web3 = require('web3');
+const config = require('../config/config.js');
+// Services
+const Scheduler = require('./services/Scheduler.js');
+const log4js = require('log4js');
+const GasPriceFetcher = require('./lib/GasPriceFetcher.js');
+const GasPriceAvg = require('./lib/GasPriceAvg.js');
+
+let gasApiPollingInterval = config.gasApiRunEvery * 1000 ; // Seconds
+let avgGasPollingInterval = config.avgGasRunEvery * 1000 ; // Seconds
+let avgGasPeriodInterval = config.periodAvgGas * 1000 * 60 ; // Minutes
+let avgGasCount;
+
+let currentEthGasBasePrice;
+let currentEthGasPriceAvg;
+
+
+
+module.exports = class GasServices {
+    constructor(logger, config, Web3 = web3) {
+        this.logger = logger;
+        this.config = config;
+        
+        this.gasPriceFetcher = new GasPriceFetcher(
+            log4js.getLogger('ETH-MAINNET-GasPriceFetcher'),
+        );
+        
+        this.gasPriceAvg = new GasPriceAvg(
+            log4js.getLogger('ETH-MAINNET-GasPriceFetAvg'),
+        );
+    }
+    
+    async startGasServices() {
+        let etherScanGasPriceService = new Scheduler(gasApiPollingInterval, logger, { run: () => runGasPriceService() });
```

Comment:
Needs `this` in `() => this.runGasPriceService()` to work properly

Reply:
Done